flusher.go

```diff
@@ -19,14 +19,17 @@ import (
 // Flush takes the slices of metrics, combines then and marshals them to json
 // for posting to Datadog.
 func (s *Server) Flush(interval time.Duration, metricLimit int) {
+	if s.IsLocal() {
+		s.FlushLocal(interval, metricLimit)
+	} else {
+		s.FlushGlobal(interval, metricLimit)
+	}
+}
+
+func (s *Server) FlushGlobal(interval time.Duration, metricLimit int) {
```

Comment:
Ugh, this is biting me again: I'm having trouble with the logic in these methods and what each means.

From a quick pass I can't remember why FlushGlobal is writing out metrics_flushed_total for metric_type:local_histogram. Shouldn't that be in FlushLocal? This leads me to believe that we need a better comment on this method that helps explain the distinction…